### PR TITLE
PopUp for AlertRule Name,  State color change, Validation "for"

### PR DESCRIPTION
**Pull request by 602dhruviii**
# Description
Added Popup for Alert Name Input on Logs Page,
Added UI: Colored Backgrounds for Alert States,
Added Validation for :- [Alert] 'for' value should be less than evaluation interval - UI

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: static/js/all-alerts.js [hunk 1]
```diff
@@ -53,6 +53,25 @@ function getAllAlerts(){
         displayAllAlerts(res.alerts);
     })
 }
+// Custom cell renderer for State field
+function stateCellRenderer(params) {
+    let state = params.value;
+    let color;
+    switch (state) {
+        case 'Normal':
+            color = '#53DB6E';
```
Comment (nkunal) [inline static/js/all-alerts.js:62]: dont use hard coded number, use color names, also dark/light theme might not work with this

Comment (602dhruviii) [reply static/js/all-alerts.js:62]: they are the colors for state of the Alert ...will that not be same throughout for light as well as dark theme?

**Comment by sonamgupta21** [comment]
Make the CSS for this popup and the overlay similar to image attached in github-issue https://github.com/siglens/siglens/issues/84
image

File: static/js/alert.js [hunk 2]
```diff
@@ -66,7 +66,34 @@ let mapIndexToAlertState=new Map([
...
+        title: 'Evaluate for interval must be greater than evaluation interval',
+        placement: 'top',
+        trigger: 'manual' 
+    });
+    $('#evaluate-for').on('input', function () {
+        let evaluateEveryValue = parseInt($('#evaluate-every').val());
+        let evaluateForValue = parseInt($(this).val());
+        let submitbtn=$("#save-alert-btn");
+
+        if (evaluateForValue > evaluateEveryValue) {
```
Comment (sonamgupta21) [inline static/js/alert.js:86]: 1. The condition should be `>=` .
2. Make the default value for `for evaluation` as `0`.
3. As you update the `evaluate every` value and it becomes greater than the `for` value the tooptip should get remove automatically.
image

File: static/index.html [hunk 3]
```diff
@@ -423,6 +423,26 @@ <h3 class="header mb-2">Add panel to dashboard</h3>
                 <button type="button" id="create-panel" class="btn create-db">Create Panel</button>
             </div>
         </div>
+        <div class="addrulepopupOverlay">
+            <div class="addrulepopupContent">
+                <h3 class="header" style="font-size: 18px;text-align: left;font-family: 'DINpro';font-weight: 600;">Create Alert</h3>
```
Comment (sonamgupta21) [inline static/index.html:428]: 1. Avoid defining styles directly in HTML.
2. Update the CSS for dark mode.
image

3. Prevent creating alerts with empty rule names.
4. Remove any transformations or delays when popups open.
5. Adjust the height of the header and buttons to match our UI. Refer to other popups for reference.

File: static/css/siglens.css [hunk 4]
```diff
@@ -27,6 +30,10 @@
     --fw-500: 500;
     --fw-900: 900;
 
+    --color-normal: #40C463;
```
Comment (sonamgupta21) [inline static/css/siglens.css:33]: Since you've added new colors, please remove the previous ones.

File: static/alert.html [hunk 5]
```diff
@@ -208,7 +208,7 @@
                                         If it is pending for longer than the "for" value, it will become a firing alert.'>
                                         </i>
                                     </span>
-                                    <input type="number" name="evaluate-every" id="evaluate-for" value="1" min="0" tabindex="5"
+                                    <input type="number" name="evaluate-every" id="evaluate-for" value="0" min="0" tabindex="5"
```
Comment (nkunal) [inline static/alert.html:211]: if you use this default value of 0 for `evaluate for` then the alert is not savable, so undo this change

Comment (602dhruviii) [reply static/alert.html]: done

File: static/index.html [hunk 6]
```diff
@@ -423,6 +423,29 @@ <h3 class="header mb-2">Add panel to dashboard</h3>
...
+                </div>
+                <div class="btncontainer">
+                    <button type="button" id="addrule-cancel-btn">Cancel</button>
+                    <button type="button" id="addrule-save-btn">Add
+                        <img src="assets/new-tab-white-icon.svg">
+                    </button>
+                </div>
+            </div>
+        
+        <div class="popupContent" id="error-popup">
```
Comment (sonamgupta21) [inline static/index.html:442]: Remove this popup.

**Review by sonamgupta21 — APPROVED** [review]
Looks good.
